models/dao/demographics.dao.js

```diff
@@ -0,0 +1,154 @@
+        return demographic.value;
+    } else if (type === 'year') {
+        return demographic.value;
+    } else if (type === 'bool') {
+        if (demographic.value === 'true') {
+            return true;
+        }
+        return false;
+    } else if (type === 'date') {
+        return demographic.value;
+    }
+    // // FIXME: only returns a true value... need to join with questionChoice
+    // else if(type === 'choice') {
+    //     return demographic.value;
+    // }
+    // // FIXME will always be null... need to join with questionChoice
+    // else if(type === 'choices') {
+    //     return demographic.value;
+    // }
+
+    return demographic.value;
+};
+
+const formatAndMergeDemographics = (demographics, questionTextObjs) => {
+    let formattedDemographics = demographics.map((demographic) => {
+        const formattedDemographic = {
+            userId: demographic['user.id'],
+        };
+        const demographicKeyText = questionTextObjs.find(textObj => textObj.id === demographic['question.id']).text;
+        formattedDemographic[demographicKeyText] = castAnswerValueByType(demographic);
+        formattedDemographic.registrationDate =
+            moment(demographic['user.createdAt'], 'YYYY-MM-DD').format('YYYY-MM-DD');
+        return formattedDemographic;
+    });
+    formattedDemographics = _.chain(formattedDemographics)
+        .groupBy('userId')
+        .map((userRecordSet) => {
+            let unifiedRecord = {};
+            userRecordSet.forEach((record) => {
+                unifiedRecord = Object.assign(unifiedRecord, record);
+            });
+            delete unifiedRecord.userId;
+            delete unifiedRecord.type;
+            return unifiedRecord;
+        })
+        .flattenDeep()
+        .value();
+    return formattedDemographics;
+};
+
+module.exports = class DemographicsDAO extends Base {
+    listDemographics(options = {}) { // TODO: orderBy query param?
+        let role = options.role ? options.role : { [Op.in]: ['clinician', 'participant'] };
+        if (role === 'all') {
+            role = { [Op.in]: ['admin', 'clinician', 'participant'] };
+        }
+        const where = { role };
+        return this.db.ProfileSurvey.findAll({
+            raw: true,
+            attributes: [
+                'id',
+            ],
+        })
+        .then((surveys) => {
+            const surveyIds = surveys.map(survey => survey.id);
+            return this.db.SurveyQuestion.findAll({
+                raw: true,
+                attributes: [
+                    'id',
+                ],
+                where: {
+                    surveyId: surveyIds,
+                },
+            })
+            .then((surveyQuestions) => {
+                const questionIds = surveyQuestions.map(surveyQuestion => surveyQuestion.id);
+                return this.db.QuestionText.findAll({
+                    raw: true,
+                    attributes: [
+                        'id',
```

Comment:
Similarly you need questionId here as discussed. There are cases when id here will not be equal to questionId. For example if you change the text of a question. Or if you add a translation of a question and after that point the surveyId and questionId will diverge.